Improve docs (#1242)
imgbot/Imgbot: Docs/contributing.md

The following is where you can find out how to get set up to run locally as well as detailed information on exactly how Imgbot works.

### Imgbot Service

The core of Imgbot runs on a serverless stack called [Azure Functions](https://azure.microsoft.com/en-us/services/functions/).
The Function apps are running the image compression, pushing the commits, and opening the Pull Requests.
Once you get the tools you need to work with Azure functions, you can run the apps locally.

You can either get the tools [integrated with Visual Studio](https://blogs.msdn.microsoft.com/webdev/2017/05/10/azure-function-tools-for-visual-studio-2017/) and use `F5` or you can [get the CLI](https://github.com/Azure/azure-functions-cli) standalone and use `func run Imgbot.Function`.
If you are using Visual Studio for Mac, there is [built-in support](https://docs.microsoft.com/en-us/visualstudio/mac/azure-functions) for Azure functions.

We also have support for running with VS Code. You will still need to get the CLI as mentioned above and the C# extension for VS Code to compile and get intellisense.

> Each function has a task you can execute that will clean + build + run the process. To start one, open the prompt with `cmd/ctrl + shift + p` and select `Run task`. From there you will see all the tasks checked into `.vscode/tasks.json`. Choose a function to run, such as `Run CompressImagesFunction`, and it will build and start up. To attach to this process choose the `Debug a function` configuration from the debug tab to see the running processes. Type `func` into the picker to see your running function and select it. It's a two-step process, the debugger and the function process. When you kill the debugger, the process will still be running. You can kill the function host by bringing up the prompt again with `cmd/ctrl + shift + p` and select `Kill the active terminal instance`.

Author: shorter?

Author: ```suggestion
We support running VS Code. You will still need to get the CLI (mentioned above) and the VS Code C# extension to compile and get IntelliSense.
Each function has an executable task that will clean + build + run the process. To start a task, open the prompt with `cmd/ctrl + shift + p` and select `Run task`. You will see all tasks checked into `.vscode/tasks.json`. Choose a function to run, such as `Run CompressImagesFunction`, and it will build and start-up. To attach to this process, choose the `Debug a function` configuration from the debug tab to see the running processes. Type `func` into the picker to see your running function and select it. The two-step process runs the debugger and the function process. When you kill the debugger, the function process will still be running. You can kill the function host by bringing up the prompt again with `cmd/ctrl + shift + p` and select `Kill the active terminal instance`.
```

The username to accompany the installation token password is `x-access-token`.

For security reasons, we cannot provide contributors with a pem file because this is a secret that delegates permissions in GitHub. You can run every part of the function except parts where authentication is required without this secret. If you are working on a part of the function that requires this secret, you can generate one for yourself to test with. [Register a GitHub app](https://github.com/settings/apps/new) for development purposes and install this app into the repo you are using to test with. Set the AppId in the code and you should be good to go.

> If there is a part of this process that isn't clear or you have any questions at all, feel free to open an issue in this repo and we'll work it out :)

Author: shorter version?

Author: ```suggestion
For unclear parts of this process or for general questions, submit an issue in this repo and we'll work it out :)
```
